File: src/main/java/me/zeroeightsix/kami/module/modules/misc/NoEntityTrace.java
package me.zeroeightsix.kami.module.modules.misc;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;

/**
 * Created by 086 on 8/04/2018.
 */
@Module.Info(name = "NoEntityTrace", category = Module.Category.MISC, description = "Blocks entities from stopping you from mining")
public class NoEntityTrace extends Module {

    @Setting(name = "Mode") private TraceMode mode = TraceMode.DYNAMIC;

    private static NoEntityTrace INSTANCE;

    public NoEntityTrace() {
        NoEntityTrace.INSTANCE = this;
    }

    public static boolean shouldBlock() {
        return INSTANCE.isEnabled() && (INSTANCE.mode == TraceMode.STATIC || mc.playerController.isHittingBlock);
    }

    private enum TraceMode {
        STATIC,DYNAMIC
    }
}

File: src/main/kotlin/org/kamiblue/client/module/modules/player/NoEntityTrace.kt
package org.kamiblue.client.module.modules.player

import net.minecraft.item.ItemPickaxe
import org.kamiblue.client.module.Category
import org.kamiblue.client.module.Module

internal object NoEntityTrace : Module(
    name = "NoEntityTrace",
    category = Category.PLAYER,
    description = "Blocks entities from stopping you from mining"
) {
    private val sneakTrigger = setting("Sneak Trigger", false)
    private val pickaxeOnly = setting("Pickaxe Only", true)

    fun shouldIgnoreEntity() = isEnabled && (!sneakTrigger.value || mc.player?.isSneaking == true)
        && (!pickaxeOnly.value || mc.player?.heldItemMainhand?.item is ItemPickaxe)
}